### PR TITLE
Introduced 'fetchBBPDevpi' function

### PR DESCRIPTION
**Pull request by arsenius7**
* using it for 'brainbuilder', 'voxcell' packages

### CODE REVIEW

File: patches/additionalPythonPackages/default.nix [hunk 1]
```diff
@@ -49,6 +49,12 @@ in
 
                             in  allPythonRecDep;
 
+  # function to fetch source tarball from BBP devpi server, given package name / version / sha256
+  fetchBBPDevpi = { pname, version, sha256 }:
+    let
+      url = "https://bbpteam.epfl.ch/repository/devpi/bbprelman/release/+f/${builtins.substring 0 3 sha256}/${builtins.substring 3 13 sha256}/${pname}-${version}.tar.gz";
```
Comment (arsenius7) [inline patches/additionalPythonPackages/default.nix]: unfortunately, sha256 makes a part of the only direct URL to source tarball that I have found

**Comment by adevress** [comment]
Hi Arseny,

Well done, I like that.

Just a little question : All the modules under devpi are immutables ? I mean they are not re-generated / with changing checksum right ?

**Comment by arsenius7** [comment]
The ones at bbprelman/release/ are immutable.

Alas, it seems that URL like +f/sha256[:3]/sha256[3:16] is an implementation detail of devpi.
Also, I guess having sha256 in the URL path is not very convenient (one have to provide a correct checksum right away instead of relying on Nix message).

In short, if there is a better way to ensure consistent version names between devpi-distributed packages and Nix-ones (which was the goal of this change), I'm open to alternatives.

**Comment by adevress** [comment]
>  Also, I guess having sha256 in the URL path is not very convenient (one have to provide a correct checksum right away instead of relying on Nix message).

if the information is available through a REST API in devpi, we might provide a little tool to help people to update later. But I like the idea anyway, that will make life of python devs easier.

It something good for pure python packages.

**Pull request merged**